src/marks/line.js

```diff
@@ -88,8 +90,25 @@ export class Line extends Mark {
                   .x((i) => X[i])
                   .y((i) => Y[i])
           )
-      )
-      .node();
+      );
+
+    if (this.halo) {
+      // With variable aesthetics, we need to regroup segments by line
```

Comment:
This is hard to follow. I think it would be clearer to initialize the DOM the desired way from the beginning, rather than creating it once (above using groupIndex) and then moving things around afterwards to apply the halo.